### ISSUE TITLE
ERROR: No objects to concatenate

### ISSUE THREAD

**Issue by tobiasko** [MannLabs/alphadia#208]
**Describe the bug**

I am trying to run alphadia throughCLI using a single diaPASEF file (generated on timsTOF Flex) and the example specLib. If fails using the default parameter file. Is this expected? Are there criticial parameters that need to be changed? Or is the problem the library (generated from ASTRAL data)? The same data was searched already by DIA-NN library-fee mode without a problem.

```
alphadia --output out --file 20240530_001_S569409_HeLaProteinDigest_5ug.d --library hela_hybrid.hdf
Removed 0 of 1 files.
0:00:00.000211 PROGRESS: Searching 1 files:
0:00:00.000314 PROGRESS:   20240530_001_S569409_HeLaProteinDigest_5ug.d
0:00:00.000363 PROGRESS: Using library: hela_hybrid.hdf.
0:00:00.000395 PROGRESS: Using 0 fasta files:
0:00:00.000424 PROGRESS: Saving output to out.
2024-06-05 11:48:13> Downloading pretrained_models.zip ...
2024-06-05 11:48:14> The pretrained models had been downloaded in /home/tobiasko/peptdeep/pretrained_models/pretrained_models.zip
0:00:00.000289 PROGRESS:           _      _         ___ ___   _
0:00:00.000409 PROGRESS:      __ _| |_ __| |_  __ _|   \_ _| /_\
0:00:00.000446 PROGRESS:     / _` | | '_ \ ' \/ _` | |) | | / _ \
0:00:00.000475 PROGRESS:     \__,_|_| .__/_||_\__,_|___/___/_/ \_\
0:00:00.000501 PROGRESS:            |_|
0:00:00.000527 PROGRESS:
0:00:00.000557 PROGRESS: version: 1.6.1
0:00:00.000590 PROGRESS: hostname: fgcz-c-073
0:00:00.000641 PROGRESS: date: 2024-06-05 11:48:21
0:00:00.000673 PROGRESS: =================== Environment ===================
0:00:00.000702 PROGRESS: alphatims       : 1.0.8
0:00:00.000730 PROGRESS: alpharaw        : 0.4.5
0:00:00.000757 PROGRESS: alphabase       : 1.2.3
0:00:00.000785 PROGRESS: alphapeptdeep   : 1.1.9
0:00:00.000812 PROGRESS: directlfq       : 0.2.19
0:00:00.000838 PROGRESS: ===================================================
0:00:00.000878 INFO: loading default config from /home/tobiasko/miniconda3/envs/alphadia/lib/python3.9/site-packages/alphadia/constants/default.yaml
0:00:00.016580 INFO: ├──version: 1
0:00:00.016634 INFO: ├──general
0:00:00.016675 INFO: │   ├──thread_count: 10
0:00:00.016713 INFO: │   ├──reuse_calibration: False
0:00:00.016749 INFO: │   ├──reuse_quant: False
0:00:00.016784 INFO: │   ├──astral_ms1: False
0:00:00.016818 INFO: │   ├──log_level: INFO
0:00:00.016853 INFO: │   ├──wsl: False
0:00:00.016887 INFO: │   ├──mmap_detector_events: False
0:00:00.016922 INFO: │   └──use_gpu: True
0:00:00.016956 INFO: ├──library_loading
0:00:00.016991 INFO: │   └──rt_heuristic: 180
0:00:00.017024 INFO: ├──library_prediction
0:00:00.017059 INFO: │   ├──predict: False
0:00:00.017094 INFO: │   ├──enzyme: trypsin
0:00:00.017129 INFO: │   ├──fixed_modifications: Carbamidomethyl@C
0:00:00.017164 INFO: │   ├──variable_modifications: Oxidation@M;Acetyl@Protein N-term
0:00:00.017198 INFO: │   ├──max_var_mod_num: 2
0:00:00.017232 INFO: │   ├──missed_cleavages: 1
0:00:00.017265 INFO: │   ├──precursor_len
0:00:00.017304 INFO: │   │   ├──7
0:00:00.017339 INFO: │   │   └──35
0:00:00.017372 INFO: │   ├──precursor_charge
0:00:00.017406 INFO: │   │   ├──2
0:00:00.017439 INFO: │   │   └──4
0:00:00.017472 INFO: │   ├──precursor_mz
0:00:00.017505 INFO: │   │   ├──400
0:00:00.017539 INFO: │   │   └──1200
0:00:00.017571 INFO: │   ├──fragment_mz
0:00:00.017605 INFO: │   │   ├──200
0:00:00.017638 INFO: │   │   └──2000
0:00:00.017677 INFO: │   ├──nce: 25.0
0:00:00.017713 INFO: │   ├──fragment_types: b;y
0:00:00.017747 INFO: │   ├──max_fragment_charge: 2
0:00:00.017781 INFO: │   ├──instrument: Lumos
0:00:00.017816 INFO: │   └──checkpoint_folder_path: None
0:00:00.017848 INFO: ├──search
0:00:00.017884 INFO: │   ├──channel_filter: 0
0:00:00.017918 INFO: │   ├──exclude_shared_ions: True
0:00:00.017952 INFO: │   ├──compete_for_fragments: True
0:00:00.017987 INFO: │   ├──target_num_candidates: 2
0:00:00.018020 INFO: │   ├──target_ms1_tolerance: 15
0:00:00.018053 INFO: │   ├──target_ms2_tolerance: 15
0:00:00.018088 INFO: │   ├──target_mobility_tolerance: 0.04
0:00:00.018122 INFO: │   ├──target_rt_tolerance: 60
0:00:00.018161 INFO: │   ├──quant_window: 3
0:00:00.018198 INFO: │   └──quant_all: True
0:00:00.018234 INFO: ├──search_advanced
0:00:00.018269 INFO: │   └──top_k_fragments: 12
0:00:00.018302 INFO: ├──calibration
0:00:00.018337 INFO: │   ├──min_epochs: 3
0:00:00.018371 INFO: │   ├──max_epochs: 20
0:00:00.018404 INFO: │   ├──batch_size: 8000
0:00:00.018438 INFO: │   ├──recalibration_target: 200
0:00:00.018472 INFO: │   ├──final_full_calibration: False
0:00:00.018506 INFO: │   └──norm_rt_mode: linear
0:00:00.018547 INFO: ├──search_initial
0:00:00.018583 INFO: │   ├──initial_num_candidates: 1
0:00:00.018618 INFO: │   ├──initial_ms1_tolerance: 30
0:00:00.018651 INFO: │   ├──initial_ms2_tolerance: 30
0:00:00.018686 INFO: │   ├──initial_mobility_tolerance: 0.08
0:00:00.018720 INFO: │   └──initial_rt_tolerance: 240
0:00:00.018753 INFO: ├──selection_config
0:00:00.018788 INFO: │   ├──peak_len_rt: 10.0
0:00:00.018824 INFO: │   ├──sigma_scale_rt: 0.5
0:00:00.018859 INFO: │   ├──peak_len_mobility: 0.01
0:00:00.018894 INFO: │   ├──sigma_scale_mobility: 1.0
0:00:00.018928 INFO: │   ├──top_k_precursors: 3
0:00:00.018962 INFO: │   ├──kernel_size: 30
0:00:00.018996 INFO: │   ├──f_mobility: 1.0
0:00:00.019031 INFO: │   ├──f_rt: 0.99
0:00:00.019065 INFO: │   ├──center_fraction: 0.5
0:00:00.019099 INFO: │   ├──min_size_mobility: 8
0:00:00.019132 INFO: │   ├──min_size_rt: 3
0:00:00.019165 INFO: │   ├──max_size_mobility: 20
0:00:00.019199 INFO: │   ├──max_size_rt: 15
0:00:00.019233 INFO: │   ├──group_channels: False
0:00:00.019267 INFO: │   ├──use_weighted_score: True
0:00:00.019301 INFO: │   ├──join_close_candidates: False
0:00:00.019336 INFO: │   ├──join_close_candidates_scan_threshold: 0.6
0:00:00.019370 INFO: │   └──join_close_candidates_cycle_threshold: 0.6
0:00:00.019403 INFO: ├──scoring_config
0:00:00.019438 INFO: │   ├──score_grouped: False
0:00:00.019472 INFO: │   ├──top_k_isotopes: 3
0:00:00.019505 INFO: │   ├──reference_channel: -1
0:00:00.019538 INFO: │   ├──precursor_mz_tolerance: 10
0:00:00.019572 INFO: │   └──fragment_mz_tolerance: 15
0:00:00.019604 INFO: ├──multiplexing
0:00:00.019639 INFO: │   ├──enabled: False
0:00:00.019673 INFO: │   ├──target_channels: 4,8
0:00:00.019706 INFO: │   ├──decoy_channel: 12
0:00:00.019740 INFO: │   ├──reference_channel: 0
0:00:00.019773 INFO: │   └──competetive_scoring: True
0:00:00.019806 INFO: ├──fdr
0:00:00.019841 INFO: │   ├──fdr: 0.01
0:00:00.019876 INFO: │   ├──group_level: proteins
0:00:00.019909 INFO: │   ├──competetive_scoring: True
0:00:00.019943 INFO: │   ├──keep_decoys: False
0:00:00.019977 INFO: │   ├──channel_wise_fdr: False
0:00:00.020010 INFO: │   └──inference_strategy: heuristic
0:00:00.020043 INFO: ├──search_output
0:00:00.020078 INFO: │   ├──peptide_level_lfq: False
0:00:00.020112 INFO: │   ├──precursor_level_lfq: False
0:00:00.020146 INFO: │   ├──min_k_fragments: 12
0:00:00.020181 INFO: │   ├──min_correlation: 0.9
0:00:00.020214 INFO: │   ├──num_samples_quadratic: 50
0:00:00.020247 INFO: │   ├──min_nonnan: 3
0:00:00.020281 INFO: │   └──normalize_lfq: True
0:00:00.020313 INFO: ├──optimization_manager
0:00:00.020348 INFO: │   ├──fwhm_rt: 5
0:00:00.020383 INFO: │   ├──fwhm_mobility: 0.01
0:00:00.020417 INFO: │   └──score_cutoff: 0
0:00:00.020449 INFO: ├──transfer_learning
0:00:00.020483 INFO: │   ├──enabled: False
0:00:00.020516 INFO: │   ├──fragment_types: b;y
0:00:00.020550 INFO: │   ├──max_charge: 2
0:00:00.020584 INFO: │   ├──top_k_samples: 3
0:00:00.020618 INFO: │   ├──norm_delta_max: True
0:00:00.020653 INFO: │   ├──precursor_correlation_cutoff: 0.5
0:00:00.020688 INFO: │   └──fragment_correlation_ratio: 0.75
0:00:00.020720 INFO: └──calibration_manager
0:00:00.020753 INFO:     ├──0
0:00:00.020788 INFO:     │   ├──name: fragment
0:00:00.020821 INFO:     │   └──estimators
0:00:00.020854 INFO:     │       └──0
0:00:00.020890 INFO:     │           ├──name: mz
0:00:00.020925 INFO:     │           ├──model: LOESSRegression
0:00:00.020958 INFO:     │           ├──model_args
0:00:00.020999 INFO:     │           │   └──n_kernels: 2
0:00:00.021034 INFO:     │           ├──input_columns
0:00:00.021069 INFO:     │           │   └──mz_library
0:00:00.021102 INFO:     │           ├──target_columns
0:00:00.021137 INFO:     │           │   └──mz_observed
0:00:00.021170 INFO:     │           ├──output_columns
0:00:00.021204 INFO:     │           │   └──mz_calibrated
0:00:00.021238 INFO:     │           └──transform_deviation: 1e6
0:00:00.021271 INFO:     └──1
0:00:00.021306 INFO:         ├──name: precursor
0:00:00.021339 INFO:         └──estimators
0:00:00.021371 INFO:             ├──0
0:00:00.021407 INFO:             │   ├──name: mz
0:00:00.021441 INFO:             │   ├──model: LOESSRegression
0:00:00.021474 INFO:             │   ├──model_args
0:00:00.021509 INFO:             │   │   └──n_kernels: 2
0:00:00.021542 INFO:             │   ├──input_columns
0:00:00.021576 INFO:             │   │   └──mz_library
0:00:00.021609 INFO:             │   ├──target_columns
0:00:00.021643 INFO:             │   │   └──mz_observed
0:00:00.021676 INFO:             │   ├──output_columns
0:00:00.021710 INFO:             │   │   └──mz_calibrated
0:00:00.021744 INFO:             │   └──transform_deviation: 1e6
0:00:00.021777 INFO:             ├──1
0:00:00.021812 INFO:             │   ├──name: rt
0:00:00.021847 INFO:             │   ├──model: LOESSRegression
0:00:00.021879 INFO:             │   ├──model_args
0:00:00.021914 INFO:             │   │   └──n_kernels: 6
0:00:00.021947 INFO:             │   ├──input_columns
0:00:00.021981 INFO:             │   │   └──rt_library
0:00:00.022014 INFO:             │   ├──target_columns
0:00:00.022048 INFO:             │   │   └──rt_observed
0:00:00.022081 INFO:             │   └──output_columns
0:00:00.022115 INFO:             │       └──rt_calibrated
0:00:00.022147 INFO:             └──2
0:00:00.022197 INFO:                 ├──name: mobility
0:00:00.022232 INFO:                 ├──model: LOESSRegression
0:00:00.022265 INFO:                 ├──model_args
0:00:00.022299 INFO:                 │   └──n_kernels: 2
0:00:00.022332 INFO:                 ├──input_columns
0:00:00.022366 INFO:                 │   └──mobility_library
0:00:00.022399 INFO:                 ├──target_columns
0:00:00.022434 INFO:                 │   └──mobility_observed
0:00:00.022467 INFO:                 └──output_columns
0:00:00.022502 INFO:                     └──mobility_calibrated
0:00:00.022929 INFO: Running DynamicLoader
0:00:00.023010 INFO: Loading .hdf library from hela_hybrid.hdf
0:00:01.568821 INFO: Running PrecursorInitializer
0:00:01.569185 INFO: Running AnnotateFasta
0:00:01.569883 INFO: Dropping decoys from input library before annotation
0:00:01.641019 INFO: Running IsotopeGenerator
0:00:01.641397 WARNING: Input library already contains isotope information. Skipping isotope generation.
 Please note that isotope generation outside of alphabase is not supported.
0:00:01.641451 INFO: Running RTNormalization
0:00:01.641531 WARNING: Input library already contains normalized RT information. Skipping RT normalization
0:00:01.641585 INFO: Saving library to out/speclib.hdf
0:00:03.135808 INFO: Running DecoyGenerator
WARNING:root:WARNING: Temp mmap arrays were written to /tmp/temp_mmap_lhe95uci. Cleanup of this folder is OS dependant, and might need to be triggered manually! Current space: 39,189,475,328
WARNING:root:WARNING: Temp mmap arrays were written to /tmp/temp_mmap_v1oc_urc. Cleanup of this folder is OS dependant, and might need to be triggered manually! Current space: 39,189,483,520
WARNING:root:WARNING: Temp mmap arrays were written to /tmp/temp_mmap_bfu_d4j0. Cleanup of this folder is OS dependant, and might need to be triggered manually! Current space: 39,189,479,424
WARNING:root:WARNING: Temp mmap arrays were written to /tmp/temp_mmap_3x8iq7xl. Cleanup of this folder is OS dependant, and might need to be triggered manually! Current space: 39,189,475,328
WARNING:root:WARNING: Temp mmap arrays were written to /tmp/temp_mmap_zgrn3pdj. Cleanup of this folder is OS dependant, and might need to be triggered manually! Current space: 39,189,479,424
0:00:18.109685 INFO: Running FlattenLibrary
0:00:23.605586 INFO: Running InitFlatColumns
0:00:23.624332 INFO: Running LogFlatLibraryStats
0:00:23.624482 INFO: ============ Library Stats ============
0:00:23.624527 INFO: Number of precursors: 513,303
0:00:23.708669 INFO: 	thereof targets:256,667
0:00:23.708845 INFO: 	thereof decoys: 256,636
0:00:23.712119 INFO: Number of elution groups: 256,667
0:00:23.712166 INFO: 	average size: 2.00
0:00:23.730511 INFO: Number of proteins: 13,582
0:00:23.732573 INFO: Number of channels: 1 ([0])
0:00:23.732727 INFO: Isotopes Distribution for 4 isotopes
0:00:23.732770 INFO: =======================================
0:00:23.734652 PROGRESS: Starting Search Workflows
0:00:23.734765 PROGRESS: Loading raw file 1/1: 20240530_001_S569409_HeLaProteinDigest_5ug
0:00:23.734874 INFO: Creating parent folder for workflows at out/.progress
0:00:23.734997 INFO: Creating workflow folder for 20240530_001_S569409_HeLaProteinDigest_5ug at out/.progress/20240530_001_S569409_HeLaProteinDigest_5ug
0:00:23.735416 INFO: Importing data from 20240530_001_S569409_HeLaProteinDigest_5ug.d
0:00:23.735499 INFO: Using .d import for 20240530_001_S569409_HeLaProteinDigest_5ug.d
0:00:23.735552 INFO: Reading frame metadata for 20240530_001_S569409_HeLaProteinDigest_5ug.d
0:00:23.923970 INFO: Reading 16,032 frames with 147,808,814 detector events for 20240530_001_S569409_HeLaProteinDigest_5ug.d
100%|██████████████████████████████████████████████████████████████████████| 16032/16032 [00:03<00:00, 4200.02it/s]
0:00:28.035279 INFO: Indexing 20240530_001_S569409_HeLaProteinDigest_5ug.d...
0:00:28.041909 INFO: Opening handle for 20240530_001_S569409_HeLaProteinDigest_5ug.d
0:00:28.051076 INFO: Fetching mobility values from 20240530_001_S569409_HeLaProteinDigest_5ug.d
0:00:28.051453 INFO: Closing handle for 20240530_001_S569409_HeLaProteinDigest_5ug.d
0:00:28.051791 INFO: Opening handle for 20240530_001_S569409_HeLaProteinDigest_5ug.d
0:00:28.054321 INFO: Fetching mz values from 20240530_001_S569409_HeLaProteinDigest_5ug.d
0:00:28.083071 INFO: Closing handle for 20240530_001_S569409_HeLaProteinDigest_5ug.d
0:00:28.083412 INFO: Indexing quadrupole dimension
0:00:28.180804 INFO: Transposing detector events
100%|██████████████████████████████████████████████████████████████████████████████| 20/20 [00:01<00:00, 18.35it/s]
0:00:30.765761 INFO: Finished transposing data
0:00:30.768795 INFO: Successfully imported data from 20240530_001_S569409_HeLaProteinDigest_5ug.d
0:00:30.854182 INFO: Initializing CalibrationManager
0:00:30.854630 INFO: Loading calibration config
0:00:30.854744 INFO: Calibration config: [{'name': 'fragment', 'estimators': [{'name': 'mz', 'model': 'LOESSRegression', 'model_args': {'n_kernels': 2}, 'input_columns': ['mz_library'], 'target_columns': ['mz_observed'], 'output_columns': ['mz_calibrated'], 'transform_deviation': '1e6'}]}, {'name': 'precursor', 'estimators': [{'name': 'mz', 'model': 'LOESSRegression', 'model_args': {'n_kernels': 2}, 'input_columns': ['mz_library'], 'target_columns': ['mz_observed'], 'output_columns': ['mz_calibrated'], 'transform_deviation': '1e6'}, {'name': 'rt', 'model': 'LOESSRegression', 'model_args': {'n_kernels': 6}, 'input_columns': ['rt_library'], 'target_columns': ['rt_observed'], 'output_columns': ['rt_calibrated']}, {'name': 'mobility', 'model': 'LOESSRegression', 'model_args': {'n_kernels': 2}, 'input_columns': ['mobility_library'], 'target_columns': ['mobility_observed'], 'output_columns': ['mobility_calibrated']}]}]
0:00:30.854834 INFO: Calibration group :fragment, found 1 estimator(s)
0:00:30.854963 INFO: Calibration group :precursor, found 3 estimator(s)
0:00:30.855076 INFO: Initializing OptimizationManager
0:00:30.855186 INFO: initial parameter: fwhm_rt = 5
0:00:30.855259 INFO: initial parameter: fwhm_mobility = 0.01
0:00:30.855357 INFO: initial parameter: score_cutoff = 0
0:00:30.855469 PROGRESS: Initializing workflow 20240530_001_S569409_HeLaProteinDigest_5ug
0:00:30.855604 INFO: Initializing OptimizationManager
0:00:30.855644 INFO: initial parameter: current_epoch = 0
0:00:30.855677 INFO: initial parameter: current_step = 0
0:00:30.855709 INFO: initial parameter: ms1_error = 30
0:00:30.855739 INFO: initial parameter: ms2_error = 30
0:00:30.855769 INFO: initial parameter: rt_error = 240
0:00:30.855819 INFO: initial parameter: mobility_error = 0.08
0:00:30.855852 INFO: initial parameter: column_type = library
0:00:30.855883 INFO: initial parameter: num_candidates = 1
0:00:30.855914 INFO: initial parameter: recalibration_target = 200
0:00:30.855947 INFO: initial parameter: accumulated_precursors = 0
0:00:30.855975 INFO: initial parameter: accumulated_precursors_01FDR = 0
0:00:30.856002 INFO: initial parameter: accumulated_precursors_001FDR = 0
0:00:30.856052 INFO: Initializing FDRManager
0:00:30.856118 INFO: Loading classifier store from /home/tobiasko/miniconda3/envs/alphadia/lib/python3.9/site-packages/alphadia/constants/classifier
0:00:30.877668 PROGRESS: Applying channel filter using only: [0]
0:00:30.955914 PROGRESS: 215,312 target precursors potentially observable (41,355 removed)
0:00:31.044211 INFO: === checking if epoch conditions were reached ===
0:00:31.044570 INFO: ❌ ms1_error      : 30.0000 > 15
0:00:31.044663 INFO: ❌ ms2_error      : 30.0000 > 15
0:00:31.044749 INFO: ❌ rt_error       : 240.0000 > 60
0:00:31.044837 INFO: ❌ mobility_error : 0.0800 > 0.04
0:00:31.044922 INFO: ❌ current_epoch  : 0 < 3
0:00:31.044999 INFO: ==============================================
0:00:31.052907 INFO: calibration group: precursor, predicting mz
0:00:31.053017 WARNING: mz prediction was skipped as it has not been fitted yet
0:00:31.053063 INFO: calibration group: precursor, predicting rt
0:00:31.053142 WARNING: rt prediction was skipped as it has not been fitted yet
0:00:31.053190 INFO: calibration group: precursor, predicting mobility
0:00:31.053257 WARNING: mobility prediction was skipped as it has not been fitted yet
0:00:31.053313 INFO: calibration group: fragment, predicting mz
0:00:31.053387 WARNING: mz prediction was skipped as it has not been fitted yet
0:00:31.053438 PROGRESS: === Epoch 0, step 0, extracting elution groups 0 to 8000 ===
0:00:31.065464 PROGRESS: Extracting batch of 15705 precursors
0:00:34.472016 INFO: Duty cycle consists of 9 frames, 0.96 seconds cycle time
0:00:34.472280 INFO: Duty cycle consists of 945 scans, 0.00049 1/K_0 resolution
0:00:34.472329 INFO: FWHM in RT is 5.00 seconds, sigma is 1.11
0:00:34.472364 INFO: FWHM in mobility is 0.010 1/K_0, sigma is 8.58
0:00:36.414461 INFO: Starting candidate selection
100%|███████████████████████████████████████████████████████████████████████| 15705/15705 [00:35<00:00, 438.96it/s]
0:01:15.540821 ERROR: Search for 20240530_001_S569409_HeLaProteinDigest_5ug failed with error No objects to concatenate
Traceback (most recent call last):
  File "/home/tobiasko/miniconda3/envs/alphadia/lib/python3.9/site-packages/alphadia/planning.py", line 314, in run
    workflow.calibration()
  File "/home/tobiasko/miniconda3/envs/alphadia/lib/python3.9/site-packages/alphadia/workflow/peptidecentric.py", line 476, in calibration
    feature_df, fragment_df = self.extract_batch(batch_df)
  File "/home/tobiasko/miniconda3/envs/alphadia/lib/python3.9/site-packages/alphadia/workflow/peptidecentric.py", line 705, in extract_batch
    sns.histplot(candidates_df, x="score", hue="decoy", bins=100)
  File "/home/tobiasko/miniconda3/envs/alphadia/lib/python3.9/site-packages/seaborn/distributions.py", line 1416, in histplot
    p.plot_univariate_histogram(
  File "/home/tobiasko/miniconda3/envs/alphadia/lib/python3.9/site-packages/seaborn/distributions.py", line 665, in plot_univariate_histogram
    hist_metadata = pd.concat([
  File "/home/tobiasko/.local/lib/python3.9/site-packages/pandas/core/reshape/concat.py", line 372, in concat
    op = _Concatenator(
  File "/home/tobiasko/.local/lib/python3.9/site-packages/pandas/core/reshape/concat.py", line 429, in __init__
    raise ValueError("No objects to concatenate")
ValueError: No objects to concatenate
0:01:15.550676 INFO: Traceback (most recent call last):
  File "/home/tobiasko/miniconda3/envs/alphadia/lib/python3.9/site-packages/alphadia/cli.py", line 341, in run
    plan.run()
  File "/home/tobiasko/miniconda3/envs/alphadia/lib/python3.9/site-packages/alphadia/planning.py", line 358, in run
    raise e
  File "/home/tobiasko/miniconda3/envs/alphadia/lib/python3.9/site-packages/alphadia/planning.py", line 314, in run
    workflow.calibration()
  File "/home/tobiasko/miniconda3/envs/alphadia/lib/python3.9/site-packages/alphadia/workflow/peptidecentric.py", line 476, in calibration
    feature_df, fragment_df = self.extract_batch(batch_df)
  File "/home/tobiasko/miniconda3/envs/alphadia/lib/python3.9/site-packages/alphadia/workflow/peptidecentric.py", line 705, in extract_batch
    sns.histplot(candidates_df, x="score", hue="decoy", bins=100)
  File "/home/tobiasko/miniconda3/envs/alphadia/lib/python3.9/site-packages/seaborn/distributions.py", line 1416, in histplot
    p.plot_univariate_histogram(
  File "/home/tobiasko/miniconda3/envs/alphadia/lib/python3.9/site-packages/seaborn/distributions.py", line 665, in plot_univariate_histogram
    hist_metadata = pd.concat([
  File "/home/tobiasko/.local/lib/python3.9/site-packages/pandas/core/reshape/concat.py", line 372, in concat
    op = _Concatenator(
  File "/home/tobiasko/.local/lib/python3.9/site-packages/pandas/core/reshape/concat.py", line 429, in __init__
    raise ValueError("No objects to concatenate")
ValueError: No objects to concatenate

0:01:15.550746 ERROR: No objects to concatenate
0:01:24.216482 WARNING: WARNING: Temp mmap arrays were written to /tmp/temp_mmap_j5nyd_ec. Cleanup of this folder is OS dependant, and might need to be triggered manually! Current space: 39,189,504,000
```

**Expected behavior**
exit without error

**Logs**
see above

**Version (please complete the following information):**
 - Installation Type Developer, Linux (Debian)

**Additional context**
Is it possible to use the AlphapeptDeep framework for library prediction in alphapeptdia? I still don't get why I need to provide a library at all.

**Comment by tobiasko** [comment]
I also tried a different .d file (also diaPASEF). Same error:

```
0:18:49.400336 ERROR: No objects to concatenate
```

**Comment by GeorgWa** [comment]
Thank you @tobiasko for reporting the bug. The issue is that this library is only for the Astral testcase and has no ion mobility information. We definetly need a more meaningful error message.

I will add a tutorial with timsTOF data later today.

**Comment by tobiasko** [comment]
I have meanwhile tested my installation on Astral data incl. alphapepDeep prediction (for Lumos). Works without any issues.

**Comment by GeorgWa** [comment]
Hi, 
[Here](https://datashare.biochem.mpg.de/s/rFyL0WBdn7zSJop) you can find some basic HeLa test data including a diaPASEF library. Please let e know if you can process your data with this library.

This issue has two components: 
1. AlphaDIA should be able to process .d data wit empirical libraries as well as predicted libraries. We will provide a fix for the bug which should enable processing.

2. The feature free processing algorithm is very sensitive and flexible (diaPASEF, synchroPASEF, midiaPASEF, slicePASEF) but also not the fastest for large libraries. I've updated the Readme to reflect this better. We have been working towards a prototype for the first pass but have decided to focus mostly on the Astral for now.

**Comment by GeorgWa** [comment]
Closed as duplicate #176

**Comment by tobiasko** [comment]
Using your emp. specLib I process my diaPASEF data.

```
0:28:29.793352 PROGRESS: =================== Search Finished ===================
```
